mlrun/launcher/base.py

```diff
@@ -357,9 +357,13 @@ def _refresh_function_metadata(runtime: "mlrun.runtimes.BaseRuntime"):
         pass
 
     @staticmethod
-    @abc.abstractmethod
-    def verify_base_image(runtime):
-        """resolves and sets the build base image if build is needed"""
+    def prepare_image_for_deploy(runtime: "mlrun.runtimes.BaseRuntime"):
+        """
+        Check if the runtime requires to build the image.
+        If build is needed, set the image as the base_image for the build.
+        If image is not given set the default one.
+        This is only needed in the client launcher since the image is built before the run is submitted.
```

Comment:
not sure why this comment is relevant to base?
it is like you dictate the behavior on base, apply logic on client launcher only
I think the first line is relevant only. those 3 marked lines are irrelevant and should be removed